### PR TITLE
Changelog and the Upgrading guide

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: docs/User/ProjectRelated/Changelog.md [hunk 1]
```diff
@@ -41,21 +43,49 @@ higher level operations with additional dependencies.
...
+* `Streamly.Data.Array.Foreign` is deprecated. Use `Streamly.Data.Array` from
+  `streamly-core` instead.
+
 * In `Streamly.Data.Fold`:
   * `head` renamed to `one`.
   * `serialWith` renamed to `splitWith`.
-  * `variance`, and `stdDev` deprecated. Please use the
-    `streamly-statistics` package instead.
-  * `chunksOf` is renamed to `groupsOf`.
+  * `variance`, and `stdDev` deprecated. Please use the `streamly-statistics`
```
Comment (adithyaov) [inline docs/User/ProjectRelated/Changelog.md:56]: Can remove this line.

File: docs/User/ProjectRelated/Changelog.md [hunk 2]
```diff
@@ -41,21 +43,49 @@ higher level operations with additional dependencies.
...
+    package instead.
+  * `chunksOf` renamed to `groupsOf`.
+  * `last` renamed to `latest`.
+  * `foldr` renamed to `foldr'`.
+  * `drainBy` renamed to `drainMapM`.
```
Comment (adithyaov) [inline docs/User/ProjectRelated/Changelog.md:61]: Can remove this line.

File: docs/User/ProjectRelated/Changelog.md [hunk 3]
```diff
@@ -41,21 +43,49 @@ higher level operations with additional dependencies.
...
+* `Streamly.Prelude`is deprecated. Use `Streamly.Data.Stream.Prelude` and
+  `Streamly.Data.Fold` from `streamly-core` instead.
+* `Streamly.Data.Fold.Tee`is deprecated. Use `Streamly.Data.Fold` from
+  `streamly-core` instead.
+* `Streamly.Data.Array.Foreign` is deprecated. Use `Streamly.Data.Array` from
+  `streamly-core` instead.
+
 * In `Streamly.Data.Fold`:
   * `head` renamed to `one`.
   * `serialWith` renamed to `splitWith`.
```
Comment (adithyaov) [inline docs/User/ProjectRelated/Changelog.md:55]: Can remove this line.

**Comment by adithyaov** [comment]
The commit removing the deprecated functions should be removed.

**Comment by harendra-kumar** [comment]
There are lot of newly exposed functions in many modules, we should mention the modules where additions have been made.

File: docs/User/ProjectRelated/Changelog.md [hunk 4]
```diff
@@ -3,60 +3,79 @@
...
+  replacing the existing `IsStream t` or `SerialT` types. Use `fromStream`,
+  `toStream` from `Streamly.Prelude` module to adapt the types.
+* Signatures changed in `Streamly.Data.Unfold`:
   * `fromStream`
   * `replicateM`
 
+### Major Changes
+
+`Streamly.Prelude` module has been deprecated, equivalent
+functionality is covered by the `Streamly.Data.Stream`,
```
Comment (adithyaov) [inline docs/User/ProjectRelated/Changelog.md:37]: `Streamly.Data.Stream` is re-exported from `Streamly.Data.Stream.Prelude` right?

File: docs/User/ProjectRelated/Changelog.md [hunk 5]
```diff
@@ -3,60 +3,79 @@
...
+`Streamly.Data.Stream` module and the `Stream` type are meant for
+writing high-performance fused pipelines not involving explicit
+recursion. For writing code that may require recursive function calls,
+`Streamly.Data.Stream.StreamK` module and the `StreamK` type have been
+added which provide a CPS based stream implementation. `Stream` and
+`StreamK` types can be easily interconverted.
+
+The old code can be adapted to use the new modules with some changes.
+See the upgrade guide for more details on how to adapt your existing
+code to the new release.
```
Comment (adithyaov) [inline docs/User/ProjectRelated/Changelog.md:51]: Upgrading guide can be linked here.

File: docs/User/ProjectRelated/Changelog.md [hunk 6]
```diff
@@ -3,60 +3,79 @@
...
+
 ### Enhancements
 
 * Added the following new modules to the `streamly-core` package:
-  * Streamly.Data.Array.Mut
+  * Streamly.Data.Array
```
Comment (adithyaov) [inline docs/User/ProjectRelated/Changelog.md:56]: The following needs to be changed. We changed the module hirerchy.

File: docs/User/ProjectRelated/Upgrading.md [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+```
+
+Earlier `consM` was used to create an implicitly concurrent stream of
+actions. In the new release, an equivalent effect is achieved by using
+a serial `consM` to create a stream of actions and then explicitly using
+`Stream.parEval` on it to evaluate it concurrently.
+
+```
+>>> consM = StreamK.consM
+>>> (|:) consM
```
Comment (adithyaov) [inline docs/User/ProjectRelated/Upgrading.md:223]:
```suggestion
>>> (|:) = consM
```

**Review by adithyaov — COMMENTED** [review]
Some comments. Looks good. I'll test the code snippets. I'll push a commit once you confirm you're done pushing your changes.

File: docs/User/ProjectRelated/Changelog.md (hunk 4 above)
Comment (harendra-kumar) [reply docs/User/ProjectRelated/Changelog.md:37]: Yes, it is.

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**